python/pyarrow/tests/test_memory.py

```diff
@@ -169,3 +178,63 @@ def test_supported_memory_backends():
         assert "jemalloc" in backends
     if should_have_mimalloc:
         assert "mimalloc" in backends
+
+
+def run_debug_memory_pool(pool_factory, env_value):
+    """
+    Run a piece of code making an invalid memory write with the
+    ARROW_DEBUG_MEMORY_POOL environment variable set to a specific value.
+    """
+    code = f"""if 1:
```

Comment:
is the `if 1` just so the string contents don't have to be dedented?

Reply:
Yes!